WcaOnRails/app/controllers/database_controller.rb

```diff
@@ -4,38 +4,41 @@ class DatabaseController < ApplicationController
   RESULTS_README_TEMPLATE = 'database/public_results_readme'
 
   def results_export
-    @sql_filename, @sql_filesize = _link_info DbDumpHelper::RESULTS_EXPORT_SQL_PERMALINK
-    @tsv_filename, @tsv_filesize = _link_info DbDumpHelper::RESULTS_EXPORT_TSV_PERMALINK
+    @sql_path, @sql_filesize = get_current_export("sql")
+    @tsv_path, @tsv_filesize = get_current_export("tsv")
+    @sql_filename = File.basename(@sql_path)
+    @tsv_filename = File.basename(@tsv_path)
 
-    @sql_rel_path = DatabaseController.rel_download_path DbDumpHelper::RESULTS_EXPORT_FOLDER, @sql_filename
-    @tsv_rel_path = DatabaseController.rel_download_path DbDumpHelper::RESULTS_EXPORT_FOLDER, @tsv_filename
-
-    @sql_perma_path = DatabaseController.rel_download_path DbDumpHelper::RESULTS_EXPORT_FOLDER, DbDumpHelper::RESULTS_EXPORT_SQL_PERMALINK
-    @tsv_perma_path = DatabaseController.rel_download_path DbDumpHelper::RESULTS_EXPORT_FOLDER, DbDumpHelper::RESULTS_EXPORT_TSV_PERMALINK
+    @sql_perma_path = "#{EnvConfig.ROOT_URL}/export/results/#{DbDumpHelper::RESULTS_EXPORT_SQL_PERMALINK}"
+    @tsv_perma_path = "#{EnvConfig.ROOT_URL}/export/results/#{DbDumpHelper::RESULTS_EXPORT_TSV_PERMALINK}"
   end
 
-  def _link_info(permalink)
-    full_permalink = DbDumpHelper::RESULTS_EXPORT_FOLDER.join(permalink)
-
-    actual_filename = File.basename File.readlink(full_permalink)
-    # Manually resolve the relative link to be independent of runtime environments
-    actual_file = DbDumpHelper::RESULTS_EXPORT_FOLDER.join(actual_filename)
-
-    filesize_bytes = File.size? actual_file
-
-    [actual_filename, filesize_bytes]
+  def sql_permalink
+    url, _ = get_current_export("sql")
+    redirect_to url, status: 301, allow_other_host: true
   end
 
-  def developer_export
-    @rel_download_path = DatabaseController.rel_download_path DbDumpHelper::DEVELOPER_EXPORT_FOLDER, DbDumpHelper::DEVELOPER_EXPORT_SQL_PERMALINK
+  def tsv_permalink
+    url, _ = get_current_export("sql")
+    redirect_to url, status: 301, allow_other_host: true
   end
 
-  def self.rel_download_path(base_folder, file_name)
-    file_path = base_folder.join file_name
-    relative_path = file_path.relative_path_from DbDumpHelper::EXPORT_PUBLIC_FOLDER.parent
+  def get_current_export(type)
+    export_timestamp = Time.new(DbDumpHelper::export_metadata["export_date"])
+
+    Rails.cache.fetch("database-export-#{export_timestamp}-#{type}", expires_in: 2.days) do
```

Comment:
We generate the results export daily, so expiry of 2 days is definitely not enough.
You can use the timestamps from the cronjobs in the cache key, similar to how we do it for CAD tasks

Reply:
that's good, that means I don't have to upload the metadata as well

Reply:
Sadly the cronjob timestamp is not the same as the metadata timestamp.
2024-01-31 00:00:06 UTC vs. 2024-01-31 00:17:37 UTC for the current export. I guess this is because the cronjob timestamp is the enqueue time and the other is the actual export time. Should I make the dump_results_db use the cronjob time stamp instead?

Reply:
Hm, which cronjob timestamp are you using? They have a run_start and run_end timestamp, and one of the two should match the metadata very closely